cwms-data-api/src/main/java/cwms/cda/data/dto/Location.java

```diff
@@ -519,31 +516,14 @@ private Nation convertStringToNation(String nation) {
     }
 
     @Override
-    public void validate() throws FieldException {
-        ArrayList<String> missingFields = new ArrayList<>();
-        if (this.getName() == null) {
-            missingFields.add("Name");
-        }
-        if (this.getLocationKind() == null) {
-            missingFields.add("Location Kind");
-        }
-        if (this.getTimezoneName() == null) {
-            missingFields.add("Timezone ID");
-        }
-        if (this.getOfficeId() == null) {
-            missingFields.add("Office ID");
-        }
-        if (this.getHorizontalDatum() == null) {
-            missingFields.add("Horizontal Datum");
-        }
-        if (this.getLongitude() == null) {
-            missingFields.add("Longitude");
-        }
-        if (this.getLatitude() == null) {
-            missingFields.add("Latitude");
-        }
-        if (!missingFields.isEmpty()) {
-            throw new RequiredFieldException(missingFields);
-        }
+    protected void validateInternal(CwmsDTOValidator validator) {
+        super.validateInternal(validator);
+        validator.required(getName(), "name");
```

Comment:
It would be nice if the validation could find the getter from the property name.  Or if it could guess the property name from the name of the getter.   Or even find the required properties from just the annotations...